compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -874,6 +892,50 @@ object Erasure {
             outerParamDefs(constr)
       else Nil
 
+    /** For all statements in stats: given a retained inline method and
+     *  its retainedBody method such as
+     *
+     *     inline override def f(x: T) = body1
+     *     private def f$retainedBody(x: T) = body2
```

Comment:
In the current implementation, would it type check if we call this method in source?

Reply:
You can't call it in source. `f$retainedBody` is a semantic name.